compiler/tests/org/jetbrains/kotlin/cli/PlatformDeclarationClashTest.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.cli
+
+import org.jetbrains.kotlin.cli.common.ExitCode
+import org.jetbrains.kotlin.cli.jvm.K2JVMCompiler
+import org.jetbrains.kotlin.config.LanguageFeature
+import org.jetbrains.kotlin.test.TestCaseWithTmpdir
+import java.io.File
+
+class PlatformDeclarationClashTest : TestCaseWithTmpdir() {
```

Comment:
1) Let's use the standard `AbstractCliTest` for this (`compiler/testData/cli/jvm`), and check that the file is absent using the `ABSENT` directive (e.g. https://github.com/JetBrains/kotlin/blob/049b6899b729fd4c04ab4588743404150525019c/compiler/testData/cli/jvm/jvmIrByDefault1_5.test#L1)
2) I'd prefer to avoid using unstable language features, let's make the test cause platform declaration clash via `@JvmName`
3) Please add a copy of the test with K2 (`-Xuse-k2`), because it has another entrypoint which also needs to be checked